### PR TITLE
2056: Migrate checkbox to mui

### PR DESCRIPTION
**Pull request by f1sh1918**
### Short Description

Migrated missing checkbox to MUI.
I think in the `primitive-inputs` is still too much application form related code, thats why i even create a more basic and reuseable component

### Proposed Changes

- create reusable checkbox base component (without form logic)
- use it in the `CheckboxForm` and in `CardSelfServiceForm` 
- move `FormAlert` to mui/base
- create unit tests for base components
- replace spinner

### Side Effects

Some error messages look different in the application form. We should decide on a particular format (with or without the icon)

image

### Testing

1. Open card creation self service portal for koblenz. Test the checkbox works and the error message will be shown
2. Open application form and test the checkboxes there

### Resolved Issues

Fixes: #2056

### CODE REVIEW

File: administration/src/mui-modules/base/__tests__/FormAlert.test.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+describe('FormAlert', () => {
+  it('should not render when errorMessage is null', () => {
+    const { container } = render(<FormAlert errorMessage={null} />)
+    expect(container.firstChild).toBeNull()
+  })
+
+  it('should not render when errorMessage is undefined', () => {
+    const { container } = render(<FormAlert errorMessage={undefined} />)
+    expect(container.firstChild).toBeNull()
+  })
```
Comment (bahaaTuffaha) [inline administration/src/mui-modules/base/__tests__/FormAlert.test.tsx:15]: Do we need to test both (null and undefined)? both are falsy anyway. 🤔

**Review by bahaaTuffaha — APPROVED** [review]
Tested on Firefox and chromium based browser ✅

Does these other check-boxes needs replacement or it's just for card creation?

image

File: administration/src/mui-modules/base/__tests__/FormAlert.test.tsx (hunk 1 above)
Comment (f1sh1918) [reply administration/src/mui-modules/base/__tests__/FormAlert.test.tsx:15]: you are right. i will change that

**Comment by f1sh1918** [comment]
> #2056

Well the root of this ticket was to migrate koblenz related stuff. i reused then the checkbox for application process to have all end user facing checkbox same style. I would adjust the other checkboxes in a separate ticket

**Review by seluianova — APPROVED** [review]
looks good to me. nit-pick fix is up to you

**Comment by f1sh1918** [comment]
Updated the missing alert old styled alert (with background)
I kept the small glitch in the alignment under the radio buttons, since we may have to override the radio buttons stylingand its really minor
